plugins/http_plugin/beast_http_session.hpp

```diff
@@ -0,0 +1,576 @@
+#if BOOST_VERSION < 107000                
+        auto& lowest_layer = beast::get_lowest_layer<tcp_socket_t&>(socket);
+#else
+        auto& lowest_layer = beast::get_lowest_layer(socket);
+#endif                
+        auto local_endpoint = lowest_layer.local_endpoint();
+        auto local_socket_host_port = local_endpoint.address().to_string() 
+                + ":" + std::to_string(local_endpoint.port());
+        const auto& host_str = req["Host"].to_string();
+        if (host_str.empty() 
+            || !host_is_valid(*plugin_state, 
+                                host_str, 
+                                local_socket_host_port, 
+                                is_conn_secure)) 
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    // Handle HTTP conneciton using boost::beast for TCP communication
+    // Subclasses of this class (plain_session, ssl_session, etc.) 
+    // use the Curiously Recurring Template Pattern so that
+    // the same code works with both SSL streams, regular TCP sockets and UNIX sockets
+    template<class Derived> 
+    class beast_http_session :  public detail::abstract_conn
+    {
+        protected:
+            beast::flat_buffer buffer_;
+
+            // time points for timeout measurement and perf metrics 
+            steady_clock::time_point session_begin_, read_begin_, handle_begin_, write_begin_;
+            uint64_t read_time_us_ = 0, handle_time_us_ = 0, write_time_us_ = 0;
+
+            // HTTP parser object
+            std::optional<http::request_parser<http::string_body>> req_parser_;
+
+            // HTTP response object
+            std::optional<http::response<http::string_body>> res_;
+
+            std::shared_ptr<http_plugin_state> plugin_state_;
+
+            // whether response should be sent back to client when an exception occurs
+            bool is_send_exception_response_ = true; 
+            
+            std::shared_ptr<eosio::chain::named_thread_pool> thread_pool_;
+
+            template<
+                class Body, class Allocator>
+            void
+            handle_request(
+                http::request<Body, http::basic_fields<Allocator>>&& req)
+            {
+                res_->version(req.version());
+                res_->set(http::field::content_type, "application/json");
+                res_->keep_alive(req.keep_alive());
+                res_->set(http::field::server, BOOST_BEAST_VERSION_STRING);
+
+                // Returns a bad request response
+                auto const bad_request =
+                [](beast::string_view why, detail::abstract_conn& conn)
+                {
+                    conn.send_response(std::string(why), 
+                                    static_cast<int>(http::status::bad_request));
+                };
+
+                // Returns a not found response
+                auto const not_found =
+                [](beast::string_view target, detail::abstract_conn& conn)
```

Comment:
The only place this is called from has a string for target, change to `const std::string&`